src/main.rs

```diff
@@ -24,18 +41,12 @@ fn compile(input_file : &mut File, output_file : &mut File, options_opt : Option
 		Ok(_) => ()
 	};
 
-	let result : Result<String, caiman::frontend::CompileError> = caiman::frontend::compile_ron_definition(& input_string, options_opt);
-	match result
-	{
-		Err(why) => panic!("Parse error: {}", why),
-		Ok(output_string) =>
-		{
-			write!(output_file, "{}", output_string);
-		}
-	}
+	let result = caiman::frontend::compile_caiman
+		(& input_string, options_opt, use_assembly);
+	output_result(result, output_file);
 }
 
-fn explicate(input_file : &mut File, output_file : &mut File)
+fn explicate(input_file : &mut File, output_file : &mut Option<File>, use_assembly : bool)
```

Comment:
Can this boolean be an enum instead? We'll almost certainly want different input types eventually and it's easy to forget what true/false means when actually using these functions.

It's ok to defer this if merging is urgent.

Reply:
makes sense to me